src/Nethermind/Nethermind.Facade/BlockchainBridge.cs

```diff
@@ -291,6 +445,56 @@ private UInt256 GetNonce(Keccak stateRoot, Address address)
             return _processingEnv.StateReader.GetNonce(stateRoot, address);
         }
 
+        //Apply changes to accounts and contracts states including precompiles
+        private void ModifyAccounts(AccountOverride[] StateOverrides, IWorldState? StateProvider, IReleaseSpec? CurrentSpec)
+        {
+            Account? acc;
+
+            foreach (AccountOverride accountOverride in StateOverrides)
+            {
+                Address address = accountOverride.Address;
+                bool accExists = StateProvider.AccountExists(address);
+                if (!accExists)
+                {
+                    StateProvider.CreateAccount(address, accountOverride.Balance, accountOverride.Nonce);
+                    acc = StateProvider.GetAccount(address);
+                }
+                else
+                    acc = StateProvider.GetAccount(address);
+
+                UInt256 accBalance = acc.Balance;
+                if (accBalance > accountOverride.Balance)
+                    StateProvider.SubtractFromBalance(address, accBalance - accountOverride.Balance, CurrentSpec);
+                else if (accBalance < accountOverride.Nonce)
+                    StateProvider.AddToBalance(address, accountOverride.Balance - accBalance, CurrentSpec);
+
+                UInt256 accNonce = acc.Nonce;
+                if (accNonce > accountOverride.Nonce)
+                    StateProvider.DecrementNonce(address);
+                else if (accNonce < accountOverride.Nonce) StateProvider.IncrementNonce(address);
+
+                _multiCallProcessingEnv.Machine.SetOverwrite(StateProvider, CurrentSpec, address,
+                    new CodeInfo(accountOverride.Code), accountOverride.MoveToAddress);
+
+
+                if (accountOverride.State is not null)
+                {
+                    accountOverride.State = new Dictionary<UInt256, byte[]>();
```

Comment:
You are overriding the dictionary? So you won't apply anything.
What you probably want is to clear the state of the account.

Reply:
Updated. Also detected an issue with protocol as setting state to clean slate is not really possible...